unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/States/EditorMode/Voxels/VoxelController.cs

```diff
@@ -0,0 +1,301 @@
+    public BuildEditorMode buildEditorMode;
+    public OutlinerController outlinerController;
+    public FreeCameraMovement freeCameraMovement;
+    
+    public LayerMask groundLayer;
+
+    DecentralandEntityToEdit lastVoxelCreated;
+
+    GameObject editionGO;
+    bool mousePressed = false, isVoxelModelActivated = false,isCreatingMultipleVoxels = false; 
+    Vector3Int lastVoxelPositionPressed;
+    Vector3 lastMousePosition;
+    Dictionary<Vector3Int, VoxelPrefab> createdVoxels = new Dictionary<Vector3Int, VoxelPrefab>();
+    ParcelScene currentScene;
+    private void Start()
+    {
+        builderInputWrapper.OnMouseDown += MouseDown;
+        builderInputWrapper.OnMouseUp += MouseUp;
+ 
+    }
+
+    private void Update()
+    {
+        if (mousePressed && isVoxelModelActivated && isCreatingMultipleVoxels)
+        {
+
+            bool fillVoxels = false;
+            Vector3Int currentPosition = Vector3Int.zero;
+            VoxelEntityHit voxelHit = buildModeController.GetCloserUnselectedVoxelEntityOnPointer();
+            if (voxelHit != null && voxelHit.entityHitted.tag == "Voxel" && !voxelHit.entityHitted.IsSelected)
+            {
+                Vector3Int position = ConverPositionToVoxelPosition(voxelHit.entityHitted.rootEntity.gameObject.transform.position);
+                position += voxelHit.hitVector;
+
+                currentPosition = position;
+                fillVoxels = true;
+            }
+            else
+            {
+                RaycastHit hit;
+                UnityEngine.Ray ray = Camera.main.ScreenPointToRay(Input.mousePosition);
+
+                if (Physics.Raycast(ray, out hit, 9999, groundLayer))
+                {
+                    currentPosition = ConverPositionToVoxelPosition(hit.point);
+                    fillVoxels = true;
+
+                }
+            }
+            if (fillVoxels)
+            {
+                FillVoxels(lastVoxelPositionPressed, currentPosition);
+            }
+
+        }
+    }
+
+    public void SetSceneToEdit(ParcelScene scene)
+    {
+        currentScene = scene;
+    }
+
+    public void SetEditObjectLikeVoxel()
+    {
+        if (!mousePressed && isVoxelModelActivated)
+        {
+            VoxelEntityHit voxelHit = buildModeController.GetCloserUnselectedVoxelEntityOnPointer();
+
+            if (voxelHit != null && voxelHit.entityHitted.IsSelected) return;
+            if (voxelHit != null && voxelHit.entityHitted.tag == "Voxel")
+            {
+
+
+                Vector3 position = ConverPositionToVoxelPosition(voxelHit.entityHitted.rootEntity.gameObject.transform.position);
+                position += voxelHit.hitVector;
+                editionGO.transform.position = position;
+            }
+            else
+            {
+                RaycastHit hit;
+                UnityEngine.Ray ray = Camera.main.ScreenPointToRay(Input.mousePosition);
+
+                if (Physics.Raycast(ray, out hit, 9999, groundLayer))
```

Comment:
Move this `9999` to a `private int const RAYCAST_DISTANCE = 9999` and replace all the occurences.